app/services/spree/retail/shopify/refunder.rb

```diff
@@ -1,14 +1,16 @@
 module Spree::Retail
   module Shopify
     class Refunder
-      def initialize(credited_money:, transaction_id:, transaction_interface: nil, refunder_interface: nil, can_issue_refund_policy_klass: nil, **options)
-        @refund_reason = options[:reason]
-        @order_id = options[:order_id]
+      def initialize(credited_money:, transaction:, refunder_interface: nil, can_issue_refund_policy_klass: nil, **options)
         @credited_money = BigDecimal.new(credited_money)
-        @transaction_interface = transaction_interface || default_transaction_interface
+        @transaction = transaction
+        @refund_reason = options[:reason]
+
+        # FIXME: HOOOOOOORJ This is abominable
+        @order_id = transaction.prefix_options[:order_id]
```

Comment:
I think a monkeypatch is acceptable. There are [numerous resource][1] that implement an explicit `order_id` method. I think the `init_prefix` macro is broken and/or is bad.

[1]: https://github.com/Shopify/shopify_api/search?utf8=%E2%9C%93&q=prefix_options